src/filter_finder.py

```diff
@@ -0,0 +1,165 @@
+"""
+This script is a diagnostic tool intended for the manual determination of what color ranges can be used to threshold out all pixels corresponding to a specific skateboard in videos (or snippets of videos) in which a single skateboard is filmed under controlled, static lighting conditions.
+
+As of 04/22/2019: In order to use this script properly, the values for path, lower_chosen, and upper_chosen must be manually edited into this code.
+
+TODO: make this script configurable to run from the commandline with no need to edit code, e.g. through `python board_filter -v "board.mp4" -c "wheels"`
+
+This script was created as part of a school project, and therefore license-free.
+"""
+
+import os.path
+import cv2
+import numpy as np
+
+__author__ = "Nick"
+
+
+#Increments by which values can be changed through keyboard controls
+sinc = 1
+binc = 5
+
+path = 'We have to manually enter the path to the video file here until we add support for commandline parameters.' #TODO: make path passable from the commandline
+
+#if path leads to valid file
+if os.path.isfile(path):
+	cap = cv2.VideoCapture(path)
+	
+	#try to read first frame
+	success, frame = cap.read()
+	
+	if(success):
+		#get dimensions of this video
+		height, width, _ = frame.shape
+
+		#lower and upper bounds for color detections, to be derived using keyboard controls. The default values given here are the averages of previous trials with Steve's skateboard.
+		lower_board = np.array([82,39,80])
+		upper_board = np.array([100,155,180])
+
+		lower_wheel = np.array([10,40,0])
+		upper_wheel = np.array([29,230,255])
+
+		lower_edge = np.array([10,25,115])
+		upper_edge = np.array([25,155,255])
+
+		#decide which color range to experiment on during this run
+		lower_chosen = lower_edge
+		upper_chosen = upper_edge
+
+		#factor by which to downsample each frame
+		scale = 2
+
+		#go through all the frames of this video
+		while(success):
+			#rescale frame for faster processing
+			frame = cv2.resize(frame, (width / scale, height / scale))
+			
+			#convert frame to HSV for easier processing
+			hsv_frame = cv2.cvtColor(frame, cv2.COLOR_BGR2HSV)
+			
+			#get masks for individual board features
+			board_mask = cv2.inRange(hsv_frame, lower_board, upper_board)
```

Comment:
use some kind of data structure to simplify this repeated work

Reply:
The only thing I can think of that would consolidate the creation of individual features' masks into a single function call (that isn't on a 1:1 relationship with each call of cv2.inRange)  is to have a function that returns a list of masks, but then we'd have to deal with a bunch of garbage related to remembering in what order we supplied the upper/lower color bound arrays. Alternatively we could return it as a dictionary where the keys (np.array * np.array tuples) map to the masks, but that seems inefficient and this would run into the same problem where we still have to say `xxx_mask = mask_dict[(lower_bound, upper_bound)]` which seems like a big step backwards in terms of efficiency, readability, and compactness.
Did you have anything specific in mind here to change this?

Reply:
as the other comment, I am not saying rewrite everything, but maybe try to consider a ds option when initially writing, for next time.

I guess here it seems like classes would be better than a data structure (but then it would be a list of objects, still data structure).
any time you have like `wheel_mask`, `edge_mask`, then `extracted_wheel`, `extracted_edge`
that could be objects called `wheel` and `extracted`, with properties `mask` and `extracted`
OBJECT ORIENTATION, or something like that.

Reply:
one advantage is if you wanted to add some 4th item to filter, you would have to add all this stuff for that 4th item, instead of simply adding a new object to a list that you pass in the color bounds.